share/config.fish

```diff
@@ -195,6 +195,68 @@ function . --description 'Evaluate contents of file (deprecated, see "source")'
 	end
 end
 
+# Try to set the locale from the system configuration if we did not inherit any.
```

Comment:
I don't really want to do that because I don't want to draw too much attention to this. It's a one-shot thing - either the locale is fixed at startup or we never touch it. If we extracted it into a function, people would see it and think of running it. And then we'd be asked to allow this to run at any point during runtime, so that users can fix their misconfiguration, and I don't want to do that because really the system should be fixed.